templates/1-basic_workflow_with_config/workflow1.py

```diff
@@ -23,87 +23,109 @@
+                        # Iterate over the different frequencies
                         for key_freq, ds in ds_dict.items():
-                            if type_dict.get("floor", False):
-                                # wont be needing this when data is completely cleaned
-                                ds["time"] = ds.time.dt.floor(type_dict["floor"])
-
-                            # drop nans and stack lat/lon in 1d loc (makes code faster)
+                            # For future steps in the workflow, we can save a lot of time by stacking the spatial coordinates into a single dimension
+                            # and dropping the NaNs. This is especially useful for big datasets.
                             if type_dict.get("stack_drop_nans", False):
                                 ds = xs.utils.stack_drop_nans(
                                     ds,
                                     ds[list(ds.data_vars)[0]]
                                     .isel(time=0, drop=True)
                                     .notnull(),
                                 )
-                            # save to zarr
+                            # Prepare the filename for the zarr file, using the format specified in paths1.yml
                             path = CONFIG["paths"]["task"].format(**cur)
+                            # Save to zarr
+                            # Once again, we manually pass the arguments from the config to the function, as they will differ for each task.
                             xs.save_to_zarr(ds=ds, filename=path, **type_dict["save"])
                             pcat.update_from_ds(ds=ds, path=path)
 
+    # The next steps follow the same pattern as the previous one, so we will not comment them in detail.
+    # Typically, a workflow step will:
+    # 1) search the ProjectCatalog for the relevant datasets
+    # 2) iterate over the datasets
+    # 3) check if the processed dataset is already in the ProjectCatalog
+    # 4) set up the dask client
+    # 5) call the function that does the processing
+    #    - with arguments automatically passed from the config if consistent for all datasets
+    #    - or with arguments manually passed if they differ
+    # 6) save the processed dataset to zarr
+    # 7) update the ProjectCatalog
+
     # --- REGRID ---
     if "regrid" in CONFIG["tasks"]:
-        # get input and iter over datasets
-        input_dict = pcat.search(**CONFIG["regrid"]["inputs"]).to_dataset_dict(**tdd)
+        # Search the ProjectCatalog for the results of the previous step, then iterate over each dataset.
+        # We usually don't have to rely on search_data_catalogs anymore after the initial extraction, because the content of the ProjectCatalog is smaller and more manageable.
+        # In most cases, we can just use the search function with the 'type' and 'processing_level' attributes.
+        input_dict = pcat.search(**CONFIG["regrid"]["inputs"]).(**tdd)
```

Comment:
```suggestion
        input_dict = pcat.search(**CONFIG["regrid"]["inputs"]).to_dataset_dict(**tdd)
```

Reply:
@vindelico Those are actually needed!

Reply:
Oups! I was searching for these and I guess I replaced them. Obviously, they are needed :-)